x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -1395,14 +1406,51 @@ export function dataRecognizerFactory(
   savedObjectsClient: SavedObjectsClientContract,
   dataViewsService: DataViewsService,
   mlSavedObjectService: MLSavedObjectService,
-  request: KibanaRequest
+  request: KibanaRequest,
+  compatibleModuleType: CompatibleModule | null
 ) {
   return new DataRecognizer(
     client,
     mlClient,
     savedObjectsClient,
     dataViewsService,
     mlSavedObjectService,
-    request
+    request,
+    compatibleModuleType
   );
 }
+
+export function filterConfigs(
```

Comment:
Even though the unit test kinda documents expected results from this utility function already, for clarity, it would be good to add a comment here to explain what's happening. For example, at first glance, it's unclear to me why we have both compatibleModuleType and moduleTagFilters.

Reply:
Updated in https://github.com/elastic/kibana/pull/168721/commits/3f57d2408bdcfe73a251bdd0a13b6e1fc17818ad